build/marketplace/MarketPlace.sol

```diff
@@ -177,4 +186,9 @@ contract MarketPlace {
         require(msg.sender == a, 'sender must be authorized');
         _;
     }
+
+    modifier unpaused(uint8 p) {
+        require(!paused[p], 'markets are paused');
```

Comment:
i think the err msg should be `principal is paused` or something with the fact that its Principal based?